CMakeLists.txt

```diff
@@ -36,6 +36,13 @@ if(NOT DEFINED LIBMGBA_ONLY)
 	get_directory_property(LIBMGBA_ONLY EXCLUDE_FROM_ALL)
 endif()
 
+# feature flags
+set(MGBA_ENABLE_VIDEO_LOGGER ON CACHE BOOL "Enable video logger")
+set(MGBA_ENABLE_FILESYSTEM ON CACHE BOOL "Enable filesystem support")
+set(MGBA_ENABLE_INPUTMAP ON CACHE BOOL "Enable input mapping interface")
+set(MGBA_ENABLE_THREADING ON CACHE BOOL "Enable threading")
+set(MGBA_ENABLE_RENDERER_SOFTWARE ON CACHE BOOL "Enable software renderer")
```

Comment:
These should absolutely __not__ be cached since many of them will be unique to each output lib. Further, since they aren't cached, they shouldn't be namespaced either.